docs/DeveloperGuide.md

```diff
@@ -298,6 +298,87 @@ The following sequence diagram shows how deleting a session works:
+This is because a recurring session can be defined a session that recurs, by given intervals, and up till a last date of session. <br>
+
+Note that:
+- `SessionDate` property inherited from `Session`, now serves as the `FirstSessionDate` in `RecurringSession`
+- Creation of a `RecurringSession` with zero recurrence is not allowed by the user. <br>
+  i.e. The first and last dates cannot be the same.<br>
+  This is so that from the perspective of the user, the notion of a `Session` and `RecurringSession` will not overlap; 
+  a recurring session must strictly recur once or more.
+
+#### Add Recurring Session Feature
+The add recurring session feature allows users to add tuition sessions that recur by a given interval, at least once.
+
+##### Implementation
+![AddRecurringSessionSequenceDiagram](images/shion/AddRecurringSessionSequenceDiagram.png)
+(Note: This is a partial sequence diagram; `AddressBookParser` at the left edge is omitted for brevity)<br>
+The process of creating a recurring session and the addition of it to the model component follows the sequence diagram above,
+and is described in the following:
+1. User input previously parsed is directed to `AddRecurringSessionParser` for further parsing with `#parse(input)`. <br>
+After multiple validity checks in the `#parse(input)` method, `AddSessionCommandParser` creates 
+   an instance of an `AddRecurringSessionCommand`.
+2. `#execute` called on this `AddRecurringSessionCommand` instance calls `hasName(name)`, 
+   `hasSession(name, reccuringSessionToAdd)` and `hasOverlappingSession(recurringSessionToAdd)` methods of `Model`
+   to ensure that the addition of the recurring session will be valid. 
+3. `addSession(name, recurringSessionToAdd)` then adds the recurring session to the specific student in the model component.
+4. The result of the command execution is encapsulated as a `CommandResult` object.
+
+The multiple validity checks in `AddRecurringSessionParser#parse(input)` as mentioned in 1. ensures users are guided to
+add the compatible recurring session. Determining that a desired recurring session is compatible, and, suggesting a 
+compatible recurring session otherwise, is handled through `AddRecurringSessionParser#parse(input)`, with key checks
+depicted below.
+![AddRecurringSessionCommandParserActivityDiagram](images/shion/AddRecurringSessionCommandParserActivityDiagram.png)
+
+##### Design Considerations
+Aspect 1: Use of Inheritance for RecurringSession and related classes
+* **Alternative 1 (current choice)**: Both `RecurringSession` and `RecurringSessionCommand` classes extends 
+  `Session` and `SessionCommand` classes
+    * Pros:
+        * Promotes "don't repeat yourself" principle, given the that recurring session is a proper subtype of superclass session.
+          Enhancements to recurring session is only additive to superclass Session.
+        * Reduced complexity. Abstract classes/interfaces are not required with the simple inheritance.
```

Comment:
Maybe can also include another Pros is increased cohesion? Since we are putting all of the similar classes together.